puzzler.py

```diff
@@ -117,8 +117,20 @@ def generate_puzzles(instream, outstream, engine, variant, req_types, multipv, d
                 and (not req_types or types[0] in req_types)
                 and (max_difficulty >= total_difficulty >= min_difficulty) and (total_quality >= min_quality)):
             sm = 'sm {};'.format(pv[0]) if stm_index == 1 else ''
-            outstream.write('{};variant {};{}bm {};eval {};difficulty {};quality {:.2f};type {};pv {}\n'.format(
-                fen, current_variant, sm, pv[stm_index], evals[0], total_difficulty, total_quality, types[0], ','.join(pv)))
+            ops = ";".join("%s %s" % (k, v) for k, v in annotations.items() if k not in ("sm", "variant"))
+            operations = f';{ops}' if ops else ''
+            outstream.write('{};variant {};{}bm {};eval {};difficulty {};quality {:.2f};type {};pv {}{}\n'.format(
```

Comment:
I think this would still duplicate the bm, eval, etc. keys when repeatedly applying the puzzler.

Reply:
Oh, I totally missed that. Will fix it today.

Reply:
Thanks. No need for a late night session though. :)

Reply:
Updated it.